plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/property/OpenShiftResourceDocumentProvider.java

```diff
@@ -53,9 +51,27 @@ protected IDocument createDocument(Object element) throws CoreException {
 		if (input != null) {
 			IResource resource = input.getResource();
 			document = new Document(resource.toJson());
+			registerConnectionListener(input);
 		}
 		return document;
 	}
+	
+	
+	
+	private synchronized void registerConnectionListener(OpenShiftResourceInput input) {
+		listener = new ConnectionListener(input, this);
+		ConnectionsRegistrySingleton.getInstance().addListener(listener);
+	}
```

Comment:
I am not quite sure about lifecycle of this document provider, but in general, one instance of document provider can be used to manage documents for many elements. If this is the case, the listener for one element will be overwritten at creating another document for another element and will never be removed from connection registry.

As a variant, connection listeners can be managed by overriding methods createElementInfo(element) and disposeElementInfo(element, info) and be stored in Map<ElementInfo, ConnectionListener>